src/app/calendar/event-editor/calendar-event-editor.component.ts

```diff
@@ -0,0 +1,65 @@
+import {Component} from "@angular/core";
+import {showCalendarEventEditorModal, hideCalendarEventEditorModal} from "../../commons/dom-functions";
+import {CalEvent} from "../../models/base";
+import CalendarService from "../../services/calendar.service";
+import {DateFormatter} from "@angular/common/src/facade/intl";
+
+@Component({
+  selector: 'calendar-event-editor',
+  templateUrl: 'calendar-event-editor.component.html',
+  styleUrls: ['calendar-event-editor.component.css']
+})
+export default class CalendarEventEditorComponent {
+
+  private event: CalEvent = new CalEvent;
+
+  // save the date picked by the ng2-datetime-picker
+  private eventStart: Date;
+  private eventEnd: Date;
+
+  // show a properly formatted date as a placeholder before a date is picked
+  private start: string;
+  private end: string;
+
+  constructor(private calendarService: CalendarService) {
+
+  }
+
+  showDialog(calendarEvent?: CalEvent) {
+    this.eventStart = null;
+    this.eventEnd = null;
+    if(calendarEvent) {
+      this.event = this.clone(calendarEvent); // avoid UI changes if not saved
```

Comment:
In ES6 (and TypeScript) you can do `this.event = Object.assign({}, calendarEvent);` to clone the event.